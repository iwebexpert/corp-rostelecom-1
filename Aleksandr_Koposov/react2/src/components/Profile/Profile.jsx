import React, { useEffect, useState } from 'react'

import {
  Card,
  CardContent,
  CardActions,
  Badge,
  Icon,
  Avatar,
  TextField,
  Button,
  Dialog,
  DialogTitle,
  CircularProgress
} from '@material-ui/core'

import './Profile.scss'

export const Profile = (props) => {
  const { user, isLoading } = props
  const { id, name, age, about, avatar } = user || {}
  const [profileState, setProfile] = useState(
    {
      init: false,
      dialog: false,
      id,
      name,
      age,
      about,
      avatar
    }
  )

  useEffect(() => {
    setProfile({
      ...profileState,
      init: true,
      ...props.user
    })
  }, [])

  const onInputChange = (event) => {
    setProfile({
      ...profileState,
      [event.target.name]: event.target.value
    })
  }

  const saveChanges = () => {
    const { onSave } = props
    const { id, name, age, about, avatar } = profileState
    onSave({ id, name, age, about, avatar })
  }

  const toggleDialog = () => {
    setProfile({
      ...profileState,
      dialog: !profileState.dialog
    })
  }

  const avatarSize = {
    width: 128,
    height: 128
  }

  return (
    <Card className="profile" elevation={3}>
      <CardContent>
        <Badge
          overlap="circle"
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'right',
          }}
          badgeContent={<Icon>edit</Icon>}
          onClick={toggleDialog}
        >
          {!isLoading ? (
            <Avatar
              style={avatarSize}
              alt="Аватар пользователя"
              src={profileState.avatar}
            />
          ) : (
              <Avatar style={avatarSize}>
                <CircularProgress />
              </Avatar>
            )}
        </Badge>
        <Dialog
          onClose={toggleDialog}
          open={profileState.dialog}
          className="avatar-dialog"
        >
          <DialogTitle>Выбор аватара</DialogTitle>
          <Card>
            <CardContent>
              <TextField
                name="avatar"
                label="Ссылка на аватар (можете изменять число в конце для теста)"
                variant="outlined"
                value={profileState.avatar}
                onChange={onInputChange}
                style={{ width: '550px' }}
              />
            </CardContent>
          </Card>
        </Dialog>
        {!isLoading ?
          (
            <div>
              <TextField
                name="name"
                label="Имя"
                variant="outlined"
                value={profileState.name}
                onChange={onInputChange}
              />
              <TextField
                name="age"
                label="Возраст"
                variant="outlined"
                type="number"
                value={profileState.age}
                onChange={onInputChange}
              />
              <TextField
                name="about"
                label="О себе"
                multiline
                variant="outlined"
                value={profileState.about}
                onChange={onInputChange}
              />
            </div>
          ) : (
            <div style={{
              width: '100%',
              textAlign: 'center'
            }}>
              <CircularProgress />
            </div>
          )}
      </CardContent>
      <CardActions>
        <Button
          color="primary"
          onClick={saveChanges}
        >Сохранить</Button>
      </CardActions>
    </Card>
  )
}
