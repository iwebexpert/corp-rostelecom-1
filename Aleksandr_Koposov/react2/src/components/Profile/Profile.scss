.profile {
  & .MuiBadge-badge {
    height: 36px;
    width: 36px;
    border-radius: 50%;
    background-color: grey;
    color: white;
    cursor: pointer;
  }

  & .MuiTextField-root {
    width: 100%;
    margin: 15px 0;
  }
}

.avatar-dialog {
  & .MuiTextField-root {
    max-width: 550px;
    width: 100%;
  }
}
