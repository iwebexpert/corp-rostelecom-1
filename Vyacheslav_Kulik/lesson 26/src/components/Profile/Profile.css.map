{
"version": 3,
"mappings": "",
"sources": [],
"names": [],
"file": "Profile.css"
}
