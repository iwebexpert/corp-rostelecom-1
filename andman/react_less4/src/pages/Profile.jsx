import React from 'react';

export function ProfilePage() {
    return (
        <div>
            <h2>Profile page</h2>
            E-mail:  test@mail.ru <hr />
            Логин: Test1 <hr />
        </div>


    );
}