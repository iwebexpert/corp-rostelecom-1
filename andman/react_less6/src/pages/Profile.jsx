import React, { Component } from 'react';

export class ProfilePage extends Component {

    render() {
        const { profile } = this.props;
        //console.log(profile);
        return (
            <div>
                <h2>Profile page</h2>
        E - mail: {profile.email} <hr />
        Логин: {profile.name} < hr />
                <img src={profile.avatar} width="200px" height="200px" />
            </div >
        );
    };
}