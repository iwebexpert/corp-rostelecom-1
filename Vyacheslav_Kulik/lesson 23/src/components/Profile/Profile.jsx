import React, {Component} from 'react'
import {TextField, Button, Grid} from '@material-ui/core'
import {Link} from 'react-router-dom'

import './Profile.scss'

export class Profile extends Component {
    state = {
        author: 'Гость',
        age: ''
    }

    handleForm = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        })
    }


    render() {

        return (
            <Grid container
                  direction='column'
                  justify="center"
                  alignItems="center">
                <Grid item>
                    <TextField name='author' label="Ваше имя" value={this.state.author} onChange={this.handleForm}/>
                </Grid>
                <Grid item>
                    <TextField name='age' label="Ваш возраст" value={this.state.age} onChange={this.handleForm}/>
                </Grid>
                <Grid item>
                    <Link to='/' style={{textDecoration: 'none'}}><Button>Save</Button></Link>

                </Grid>

            </Grid>

        )
    }

}