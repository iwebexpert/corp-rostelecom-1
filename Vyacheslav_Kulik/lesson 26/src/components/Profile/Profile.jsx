import React, {Component} from 'react'
import {TextField, Button, Grid} from '@material-ui/core'
import {Link} from 'react-router-dom'

import './Profile.scss'

export class Profile extends Component {
    state = {
        author: this.props.profile.author,
        age: this.props.profile.age
    }


    handleForm = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        })
    }

    save = (event) => {
        this.props.getProfileData(this.state)
    }

    render() {


        return (
            <Grid container
                  direction='column'
                  justify="center"
                  alignItems="center">
                <Grid item>
                    <TextField name='author' label="Ваше имя" value={this.state.author} onChange={this.handleForm}/>
                </Grid>
                <Grid item>
                    <TextField name='age' label="Ваш возраст" value={this.state.age} onChange={this.handleForm}/>
                </Grid>
                <Grid item>
                    <Button onClick={this.save}>Save</Button>

                </Grid>

            </Grid>

        )
    }

}