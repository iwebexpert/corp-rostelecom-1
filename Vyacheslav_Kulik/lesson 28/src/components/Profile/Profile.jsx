import React, {useState} from 'react'
import {TextField, Button, Grid} from '@material-ui/core'
import {Link} from 'react-router-dom'

import './Profile.scss'

export const Profile = (props) => {

    const initialState = {
        author: props.profile.author,
        age: props.profile.age
    }

    const [state, setState] = useState(initialState)

    const handleForm = (event) => {
        setState({
            ...state,
            [event.target.name]: event.target.value
        })
    }

    const save = (event) => {
        props.getProfileData(state)
    }


    return (
        <Grid container
              direction='column'
              justify="center"
              alignItems="center">
            <Grid item>
                <TextField name='author' label="Ваше имя" value={state.author} onChange={handleForm}/>
            </Grid>
            <Grid item>
                <TextField name='age' label="Ваш возраст" value={state.age} onChange={handleForm}/>
            </Grid>
            <Grid item>
                <Button onClick={save}>Save</Button>

            </Grid>

        </Grid>

    )


}
//
// export class Profile extends Component {
//     state = {
//         author: this.props.profile.author,
//         age: this.props.profile.age
//     }
//
//
//     handleForm = (event) => {
//         this.setState({
//             [event.target.name]: event.target.value
//         })
//     }
//
//     save = (event) => {
//         this.props.getProfileData(this.state)
//     }
//
//     render() {
//
//
//         return (
//             <Grid container
//                   direction='column'
//                   justify="center"
//                   alignItems="center">
//                 <Grid item>
//                     <TextField name='author' label="Ваше имя" value={this.state.author} onChange={this.handleForm}/>
//                 </Grid>
//                 <Grid item>
//                     <TextField name='age' label="Ваш возраст" value={this.state.age} onChange={this.handleForm}/>
//                 </Grid>
//                 <Grid item>
//                     <Button onClick={this.save}>Save</Button>
//
//                 </Grid>
//
//             </Grid>
//
//         )
//     }
//
// }