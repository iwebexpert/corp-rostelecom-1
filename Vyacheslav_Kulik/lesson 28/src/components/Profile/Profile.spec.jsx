import React from 'react'
import {Profile} from './Profile'

describe('Profile Test', () => {
    it('should render Profile snapshot', function () {
        const container = (<Profile profile={{author: 'Test',  age: '25'}}/>)
        const wrapper = render(container)
        expect(wrapper).toMatchSnapshot()
    })
})