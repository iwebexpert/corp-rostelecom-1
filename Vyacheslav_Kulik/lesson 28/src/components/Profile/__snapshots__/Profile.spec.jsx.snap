// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Test should render Profile snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink="true"
      >
        Ваше имя
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          name="author"
          type="text"
          value="Test"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink="true"
      >
        Ваш возраст
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          name="age"
          type="text"
          value="25"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Save
      </span>
    </button>
  </div>
</div>
`;
